lib/azure/armrest/armrest_service.rb

```diff
@@ -171,25 +171,37 @@ def self.fetch_subscription_id(config)
 
         url = File.join(Azure::Armrest::RESOURCE, "subscriptions?api-version=#{config.api_version}")
 
-        response = rest_get(
-          :url         => url,
-          :proxy       => config.proxy,
-          :ssl_version => config.ssl_version,
-          :ssl_verify  => config.ssl_verify,
-          :headers => {
-            :content_type  => config.content_type,
-            :authorization => config.token
-          }
-        )
-
+        response = get_response(url, config)
         array = JSON.parse(response)["value"]
 
         if array.nil? || array.empty?
           raise ArgumentError, "No associated subscription found"
         end
 
+        results = []
+
+        # Skip over any invalid subscriptions
+        array.each do |sub_hash|
+          sub_id = sub_hash['subscriptionId']
+
+          # Use tagNames as a test URL
+          test_url = File.join(
+            Azure::Armrest::RESOURCE, 'subscriptions', sub_id,
+            "tagNames?api-version=#{config.api_version}"
+          )
+
+          begin
+            get_response(test_url, config)
+          rescue Azure::Armrest::UnauthorizedException, Azure::Armrest::BadRequestException
+            next
+          else
+            results << sub_hash
+            break if sub_hash['state'] == 'Enabled'
+          end
+        end
+
         # Look for the first enabled subscription, otherwise just take the first subscription found.
-        hash = array.find{ |h| h['state'] == 'Enabled' } || array.first
+        hash = results.find { |h| h['state'] == 'Enabled' } || results.first
 
```

Comment:
@djberg96  - Do we need this line? 
Will the results array only contain 1 subscription given the `break` statement?